src/lightning_app/core/queues.py

```diff
@@ -378,10 +397,13 @@ def get(self, timeout: int = None) -> Any:
                 pass
 
     def _get(self):
-        resp = self.client.post(f"v1/{self.app_id}/{self._name_suffix}", query_params={"action": "pop"})
-        if resp.status_code == 204:
+        try:
+            resp = self.client.post(f"v1/{self.app_id}/{self._name_suffix}", query_params={"action": "pop"})
+            if resp.status_code == 204:
+                raise queue.Empty
+            return pickle.loads(resp.content)
+        except ConnectionError:
```

Comment:
Is this error handling here  to cover the case when the instance gets taken away? If so, could you please document this here/test it? I think we should have a test that asserts Empty is raised when ConnectionError is triggered, probably also in the same place where we test for `resp.status_code == 204`

Reply:
No, this is used for bifrost. If there is an ConnectionError when trying to get a delta, this would break the flow and kill the app. Instead, we need to consider this as an empty queue.

Reply:
So this is basically unrelated to the preemptible feature? Was it a source for flakiness in the CI?

Reply:
For preemptible, we are relying on bifrost. We didn't implement this part properly. The source of flakiness is independent to those changes.

Reply:
Sounds good. I would still add a comment in the source code. With these generic errors, it is easy to forget what they are for.

Reply:
Done !